### PR TITLE
Create AuthorizedKeysCommand

### PR DESCRIPTION
**Pull request by zeripath**
This pull-request adds a subcommand that can be used as an SSH AuthorizedKeysCommand.

The configuration is described in command-line.md.

### CODE REVIEW

File: cmd/keys.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
+// Copyright 2014 The Gogs Authors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (zeripath) [reply cmd/keys.go]: Removed

**Review by techknowlogick — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]
One nit, Can you add setting to `custom/conf/app.ini.sample`

**Comment by zeripath** [comment]
@techknowlogick you mean add the `NO_AUTHORIZED_KEYS_FILE=false` setting to `custom/conf/app.ini`?

Hmm. I think if it's going to go in to the suggested `app.ini` by default, I should change it to `SSH_CREATE_AUTHORIZED_KEYS_FILE=true` so that we don't have a double negative.

**Comment by techknowlogick** [comment]
That’s correct. Your idea of no double negatives is a good one.

**Comment by zeripath** [comment]
Sorry I just noticed that XORM was reporting stuff on stdout that would conflict with the authorized_keys standard so adjusted the code.

**Review by lafriks — APPROVED** [review]

**Comment by zeripath** [comment]
I guess this request partially closes #1870?

**Comment by sapk** [comment]
@zeripath I am looking at this to use it in the docker and remove the write to authorized_keys file but I can figure why the -type flag is required since not needed and seems not used. Do you have any insight ?

File: cmd/keys.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+	if content == "" {
+		return errors.New("No key type and content provided")
+	}
+
+	if err := initDBDisableConsole(true); err != nil {
+		return err
+	}
+
+	publicKey, err := models.SearchPublicKeyByContent(content)
```
Comment (lafriks) [inline cmd/keys.go:79]: This should use internal api not models directly

Comment (zeripath) [reply cmd/keys.go:79]: That would have been a good spot >1 year ago... I guess we need to pop a bug fix up...

Comment (lafriks) [reply cmd/keys.go:79]: It just popped up in notifications :sweat_smile:

**Pull request merged**